### PR TITLE
Build: Bump gradle-baseline-java from 4.0.0 to 4.42.0

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ private boolean isDirectory(Path path) {
       return false;
     } catch (IOException e) {
       if (shouldSuppressPermissionError(e)) {
-        LOG.warn("Unable to list directory {}: {}", path, e);
+        LOG.warn("Unable to list directory {}", path, e);
```
Comment (nastra) [inline core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java:180]: this was actually producing an errorprone error

File: aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java [hunk 2]
```diff
@@ -351,7 +351,7 @@ private static void registerResource(String s3Location) {
       // when a resource is registered, LF will update SLR with necessary permissions which has a propagation delay
       waitForIamConsistency();
     } catch (AlreadyExistsException e) {
-      LOG.warn("Resource {} already registered. Error: {}", arn, e);
+      LOG.warn("Resource {} already registered. Error: {}", arn, e.getMessage(), e);
```
Comment (nastra) [inline aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java:354]: this was actually producing an errorprone error (same for the other places around logging)

Comment (kbendick) [reply aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java:354]: This is just an integration test, so not a big deal. But does it make sense to log both `e.getMessage()` along with `e`?

I don't want to change the aws integration tests too much, but if we weren't inside the aws integration tests I would suggest `LOG.warn("Resource {} already registered", arn, e)` instead.

File: aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java [hunk 3]
```diff
@@ -384,7 +384,7 @@ private static void attachRolePolicyIfNotExists(String policyArn, String policyN
       iam.getRolePolicy(GetRolePolicyRequest.builder().roleName(roleName).policyName(policyName).build());
       LOG.info("Policy {} already attached to role {}", policyName, roleName);
     } catch (NoSuchEntityException e) {
-      LOG.info("Attaching policy {} to role {} {}", policyName, roleName, e);
+      LOG.info("Attaching policy {} to role {}", policyName, roleName, e);
```
Comment (kbendick) [inline aws/src/integration/java/org/apache/iceberg/aws/lakeformation/LakeFormationTestBase.java:387]: 👍

File: build.gradle [hunk 4]
```diff
@@ -27,7 +27,7 @@ buildscript {
   }
   dependencies {
     classpath 'gradle.plugin.com.github.jengelman.gradle.plugins:shadow:7.0.0'
-    classpath 'com.palantir.baseline:gradle-baseline-java:4.0.0'
+    classpath 'com.palantir.baseline:gradle-baseline-java:4.42.0'
```
Comment (kbendick) [inline build.gradle:30]: Nit / non-blocking: Should we separate the error prone changes from the upgrade itself? Not a big deal to me.

**Review by kbendick — COMMENTED** [review]
It looks like the Java build checks failed due to a gradle memory issue. Does this upgrade bring in changes to the used memory in gradle or is this a transient issue?

```
java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOfRange(Arrays.java:3664)
	at java.lang.String.<init>(String.java:207)
	at java.io.DataInputStream.readUTF(DataInputStream.java:661)
	at java.io.DataInputStream.readUTF(DataInputStream.java:564)
	at org.gradle.internal.serialize.InputStreamBackedDecoder.readString(InputStreamBackedDecoder.java:59)
	at org.gradle.launcher.daemon.context.DefaultDaemonContext$Serializer.read(DefaultDaemonContext.java:107)
	at org.gradle.launcher.daemon.context.DefaultDaemonContext$Serializer.read(DefaultDaemonContext.java:94)
	at org.gradle.launcher.daemon.registry.DaemonInfo$Serializer.read(DaemonInfo.java:135)
	at org.gradle.launcher.daemon.registry.DaemonRegistryContent$Serializer.readInfosMap(DaemonRegistryContent.java:139)
	at org.gradle.launcher.daemon.registry.DaemonRegistryContent$Serializer.read(DaemonRegistryContent.java:119)
	at org.gradle.launcher.daemon.registry.DaemonRegistryContent$Serializer.read(DaemonRegistryContent.java:113)
	at org.gradle.cache.internal.SimpleStateCache.deserialize(SimpleStateCache.java:144)
	at org.gradle.cache.internal.SimpleStateCache.access$000(SimpleStateCache.java:34)
	at org.gradle.cache.internal.SimpleStateCache$1.create(SimpleStateCache.java:52)
	at org.gradle.cache.internal.DefaultFileLockManager$DefaultFileLock.readFile(DefaultFileLockManager.java:194)
	at org.gradle.cache.internal.OnDemandFileAccess.readFile(OnDemandFileAccess.java:43)
	at org.gradle.cache.internal.SimpleStateCache.get(SimpleStateCache.java:49)
	at org.gradle.cache.internal.FileIntegrityViolationSuppressingPersistentStateCacheDecorator.get(FileIntegrityViolationSuppressingPersistentStateCacheDecorator.java:34)
	at org.gradle.launcher.daemon.registry.PersistentDaemonRegistry.getAll(PersistentDaemonRegistry.java:71)
	at org.gradle.launcher.daemon.server.DaemonRegistryUnavailableExpirationStrategy.checkExpiration(DaemonRegistryUnavailableExpirationStrategy.java:57)
	at org.gradle.launcher.daemon.server.expiry.AnyDaemonExpirationStrategy.checkExpiration(AnyDaemonExpirationStrategy.java:43)
	at org.gradle.launcher.daemon.server.MasterExpirationStrategy.checkExpiration(MasterExpirationStrategy.java:73)
	at org.gradle.launcher.daemon.server.Daemon$DaemonExpirationPeriodicCheck.run(Daemon.java:268)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Unexpected exception thrown.
java.lang.OutOfMemoryError: Java heap space

> Task :iceberg-spark:iceberg-spark-3.2_2.12:checkstyleTest FAILED
Unexpected exception thrown.
org.gradle.internal.remote.internal.MessageIOException: Could not read message from '/127.0.0.1:47610'.
	at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:94)
	at org.gradle.internal.remote.internal.hub.MessageHub$ConnectionReceive.run(MessageHub.java:270)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.OutOfMemoryError: Java heap space
```

This LGTM, other than concerns around the memory issue when running tests. If it's just a one off, that's fine, but we might need to change the gradle memory settings or something.

**Comment by nastra** [comment]
This is low priority but I need to investigate why the OOM occurs

File: gradle.properties [hunk 5]
```diff
@@ -24,4 +24,4 @@ systemProp.knownSparkVersions=2.4,3.0,3.1,3.2,3.3
 systemProp.defaultScalaVersion=2.12
 systemProp.knownScalaVersions=2.12,2.13
 org.gradle.parallel=true
-org.gradle.jvmargs=-Xmx768m
+org.gradle.jvmargs=-Xmx1024m
```
Comment (nastra) [inline gradle.properties:27]: it seems that the plugin needs slightly more memory as it was running OOM with 768M

**Review by danielcweeks — APPROVED** [review]

**Pull request closed without merging**